tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,284 @@
+          type: Object,  // function: string => string
+          value: () => ({scale: runsColorScale}),
+        },
+        requestManager: Object,
+        _canceller: {
+          type: Object,
+          value: () => new Canceller(),
+        },
+        _attached: Boolean,
+        // The value field is a function that returns a function because Polymer
+        // will actually call the value field if the field is a function. However,
+        // we actually want the value itself to be a function.
+        _xComponentsCreationMethod: {
+          type: Object,
+          value: () => (() => {
+            const scale = new Plottable.Scales.Linear();
+            return {
+              scale: scale,
+              axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+              accessor: d => d.recall,
+            };
+          }),
+          readOnly: true,
+        },
+      },
+      observers: [
+          'reload(run, tag)',
+          '_runsChanged(_attached, runs.*)',
+          '_setChartData(_runToData, stepCapPerRun)',
+      ],
+      _computeRunColor(run) {
+        return this._colorScaleFunction.scale(run);
+      },
+      attached() {
+        // Defer reloading until after we're attached, because that ensures that
+        // the requestManager has been set from above. (Polymer is tricky
+        // sometimes)
+        this._attached = true;
+        this.reload();
+      },
+      reload() {
+        if (!this._attached) {
+          return;
+        }
+        this._canceller.cancelAll();
+        const router = getRouter();
+        let url = router.pluginRoute('pr_curves', '/pr_curves');
+        url += url.indexOf('?') > -1 ? '&' : '?';
+        const urlRunsPortion = this.runs.map(r => `run=${r}`).join('&');
+        url += `tag=${this.tag}&${urlRunsPortion}`;
+        const updateData = this._canceller.cancellable(result => {
+          if (result.cancelled) {
+            return;
+          }
+
+          const runToData = result.value;
+          this.set('_runToData', runToData);
+          this.fire('data-updated', {
+            'runToData': runToData,
+          });
+        });
+        this.requestManager.request(url).then(updateData);
+      },
+      _setChartData(runToData, stepCapPerRun) {
+        if (!runToData) {
+          return;
+        }
+
+        _.forOwn(runToData, (entries, run) => {
+          if (!entries || !entries.length) {
+            return;
+          }
+
+          const stepCap = stepCapPerRun[run];
+          let entriesIndex = entries.length - 1;
+          while (entriesIndex > 0) {
+            if (entries[entriesIndex].step <= stepCap) {
+              // This is the rightmost entry closest (or equal to) the step cap.
+              break;
+            }
+            entriesIndex--;
+          }
+          const entry = entries[entriesIndex];
+
+          // Update the step shown for this run within the card.
+          this._actualStepPerRun[run] = entry.step;
```

Comment:
There's really _no reason_ to do _any_ mutation here. Just assign `this.set('_stepCapPerRun', stepCapPerRun)`, and let `_actualStepPerRun` be a property computed from `stepCapPerRun`. What am I missing?